File: src/main/kotlin/introductory_problems/MissingNumber.java
package introductory_problems;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class MissingNumber {
    static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = Integer.parseInt(scanner.nextLine());
        List<Integer> numbers = new ArrayList<>();

        String[] input = scanner.nextLine().split(" ");
        for (String s : input) {
            numbers.add(Integer.parseInt(s));
        }

        int answer = solution(n, numbers);
        System.out.println(answer);
    }

    public static int solution(int n, List<Integer> numbers) {
        int expected;

        if (n % 2 == 1) {
            int median = (int) Math.ceil(n / 2.0);
            expected = median * (n + 1) - median;
        } else {
            expected = (n / 2) * (n + 1);
        }

        int sum = 0;
        for (int num : numbers) {
            sum += num;
        }

        return expected - sum;
    }
}

File: src/main/kotlin/cses/introductory_problems/MissingNumber.kt
import kotlin.math.ceil

fun main() {
//    val n = readln().toInt()
//    val numbers = readln().split(" ").map { it.toInt() }
    val n = 5
    val numbers = listOf(5, 2, 1, 3)

    val answer = solution(n, numbers)

    println(answer)
}

private fun solution(n: Int, numbers: List<Int>): Int {
    val expected = if (n % 2 == 1) {
        val median = ceil(n / 2.0).toInt()
        median * (n + 1) - median
    } else n / 2 * (n + 1)

    return expected - numbers.sum()
}